src/screens/view-wallet-mnemonic/view-wallet-mnemonic.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { View, Alert } from 'react-native';
+import { Text } from '../../library';
+import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';
+import { Button } from '../../library/button/button';
+
+import stylesProvider from './styles';
+import { withTheme } from '../../core/theme/with-theme';
+import { translate } from '../../core/i18n';
+import { HeaderLeft } from '../../components/header-left/header-left';
+import { smartConnect } from '../../core/utils/smart-connect';
+import { PasswordModal } from '../../components/password-modal/password-modal';
+import { hash } from '../../core/secure/encrypt';
+import { readEncrypted } from '../../core/secure/storage';
+
+export interface IProps {
+    navigation: NavigationScreenProp<NavigationState, NavigationParams>;
+    styles: ReturnType<typeof stylesProvider>;
+}
+
+interface IState {
+    mnemonic: string[];
+    showPasswordModal: boolean;
+}
+
+export const navigationOptions = ({ navigation }: any) => ({
+    headerLeft: () => {
+        return (
+            <HeaderLeft
+                icon="close"
+                text="Close"
+                onPress={() => {
+                    navigation.goBack(null);
+                }}
+            />
+        );
+    },
+    title: 'View phrase'
+});
+
+export class ViewWalletMnemonicScreenComponent extends React.Component<IProps, IState> {
+    public static navigationOptions = navigationOptions;
+
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            mnemonic: new Array(24).fill(''),
+            showPasswordModal: true
+        };
+    }
+
+    public async onEnterPassword(password) {
+        const walletId = this.props.navigation.state?.params?.wallet.id;
+
+        if (!walletId) {
+            // TODO show an error?
+            this.props.navigation.goBack(null);
+        }
+
+        try {
+            const passHash = await hash(password);
```

Comment:
i think it would be better to use `HDWallet.loadFromStorage()`, to keep this login in only one place

Reply:
`HDWallet.loadFromStorage()` does not expose mnemonic and basically uses same function

Reply:
added a getter for the above